src/cpu/aarch64/matmul/jit_bf16_matmul.cpp

```diff
@@ -0,0 +1,462 @@
+        k_residual_blk = (brg.K % (brg.k_blk * brg.rd_block)) % brg.k_blk;
+        ldb = (brg.is_n_tail) ? div_up(brg.n_tail, 4) : brg.ld_block;
+        bdb = (brg.is_m_tail) ? div_up(brg.m_tail, 2) : brg.bd_block / 2;
+
+        if (brg.is_n_tail) {
+            if (brg.n_tail % brg.n_blk == 0) {
+                pred_st = (brg.n_tail % (brg.n_blk * 2) == 0) ? sv_len
+                                                              : sv_len / 2;
+            } else {
+                pred_st = (ldb % 2 == 0)
+                        ? (sv_len / 2) + (brg.n_tail % brg.n_blk)
+                        : (brg.n_tail % brg.n_blk);
+            }
+        }
+        set_preg(prd_st.s, pred_st, X_TMP_0, X_TMP_1);
+    }
+
+    void generate() override {
+        preamble();
+        config();
+
+        LDR_IMM(reg_a, reg_param, GET_OFF(src));
+        LDR_IMM(reg_b, reg_param, GET_OFF(wei));
+        LDR_IMM(reg_c, reg_param, GET_OFF(dst));
+
+        han_blk();
+        postamble();
+    }
+
+    jit_bf16_matmul_kernel_t(
+            const dnnl::impl::cpu::aarch64::matmul::brg_bf16_t &k)
+        : brg(k) {}
+    ~jit_bf16_matmul_kernel_t() override = default;
+
+    dnnl::impl::cpu::aarch64::matmul::brg_bf16_t brg;
+
+    int ldb;
+    int bdb;
+    int rdb;
+    int k_full_blks;
+    int k_tail_blks;
+    int k_residual_blk;
+};
+
+status_t jit_bf16_matmul_t::pd_t::init(engine_t *engine) {
+    const memory_desc_wrapper src_d(src_md_);
+    const memory_desc_wrapper weights_d(weights_md_);
+    const memory_desc_wrapper dst_d(dst_md_);
+
+    brg.fp_mathmode_is_bf16
+            = one_of(this->attr()->fpmath_.mode_, fpmath_mode::bf16);
+
+    if (brg.fp_mathmode_is_bf16) {
+        weights_md_.data_type = data_type::bf16;
+        src_md_.data_type = data_type::bf16;
+    }
+
+    const bool no_runtime_dims_or_strides
+            = !(src_d.has_runtime_dims_or_strides()
+                    || weights_d.has_runtime_dims_or_strides());
+
+    VDISPATCH_MATMUL(
+            no_runtime_dims_or_strides, VERBOSE_RUNTIMEDIM_UNSUPPORTED);
+
+    const auto src_type = src_md(0)->data_type;
+    const auto wei_type = weights_md(0)->data_type;
+    const auto dst_type = dst_md(0)->data_type;
+
+    bool is_bf16 = utils::everyone_is(bf16, src_type, wei_type);
+    const bool dt_correct = (is_bf16) && utils::everyone_is(f32, dst_type);
+
+    VDISPATCH_MATMUL(dt_correct, VERBOSE_UNSUPPORTED_DT);
+    VDISPATCH_MATMUL(attr()->has_default_values(
+                             primitive_attr_t::skip_mask_t::post_ops, dst_type),
+            VERBOSE_UNSUPPORTED_ATTR);
+    VDISPATCH_MATMUL(formats_ok(), VERBOSE_UNSUPPORTED_TAG);
+    VDISPATCH_MATMUL(mayiuse(sve_256), VERBOSE_UNSUPPORTED_ISA);
+    VDISPATCH_MATMUL(!with_bias(), VERBOSE_UNSUPPORTED_BIAS_CFG);
+
+    int dims = src_d.ndims();
+
+    switch (dims) {
+        case 2: {
+            VCHECK_BG(memory_desc_init_by_tag(src_md_, format_tag::AB2a4b),
```

Comment:
So this currently supports any or AB2a4b, but often these matmuls would have at least `ab` `src`. Is there a plan to use the built in reorders from `jit_int8_matmul_t`? As far as I understand they wouldn't take much modification.